lms/static/scripts/frontend_apps/components/YouTubePicker.tsx

```diff
@@ -70,20 +80,25 @@ export default function YouTubePicker({
       ]}
     >
       <URLFormWithPreview
-        onURLChange={verifyURL}
-        onInput={resetToInitialState}
+        onURLChange={onURLChange}
+        onInput={resetCurrentUrl}
         error={error}
         inputRef={inputRef}
         urlPlaceholder="e.g. https://www.youtube.com/watch?v=cKxqzvzlnKU"
         label="Enter the URL of a YouTube video:"
         defaultURL={defaultURL}
-        thumbnail={thumbnail ?? { orientation: 'landscape' }}
+        thumbnail={{
+          isLoading: videoInfo.isLoading,
+          image: videoInfo.data?.image,
+          alt: videoInfo.data?.title,
+          orientation: 'landscape',
+        }}
       >
-        {metadata && (
+        {videoInfo.data?.title && videoInfo.data.channel && (
           <div className="flex flex-row space-x-2" data-testid="selected-video">
             <CheckIcon className="text-green-success" />
             <div className="grow font-bold italic">
-              {metadata.title} ({metadata.channel})
+              {videoInfo.data.title} ({videoInfo.data.channel})
```

Comment:
With the current metadata presentation I don't think it is completely obvious that the part in parentheses is the channel name, as opposed to part of the title.

<img width="659" alt="Channel title" src="https://github.com/hypothesis/lms/assets/2458/33488ffd-1329-4c27-b01e-efd1bcdd7b54">

For this PR we can keep the layout as-is, but I suggest we revise this in future.